sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -0,0 +1,206 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RSTMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RSTMGR_H_
+
+/**
+ * @file
+ * @brief <a href="">Reset Manager</a> Device Interface Functions
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Reset Manager generic enable/disable enumeration.
+ *
+ * Enumeration used to enable/disable bits, flags, ...
+ */
+typedef enum dif_rstmgr_enable {
+  kDifRstmgrDisable = 0, /**< disable. */
+  kDifRstmgrEnable,      /**< enable. */
+} dif_rstmgr_enable_t;
+
+/**
+ * Reset Manager peripheral software reset control.
+ */
+typedef enum dif_rstmgr_software_reset {
+  /**
+   * Simple reset (release straight away).
+   */
+  kDifRstmgrSoftwareReset = 0,
+  kDifRstmgrPeripheralResetHold,    /**< Hold peripheral in reset. */
+  kDifRstmgrPeripheralResetRelease, /**< Release peripheral from reset. */
+} dif_rstmgr_software_reset_t;
+
+/**
+ * Reset Manager bitfield encoding the reset causes information.
+ */
+typedef uint32_t dif_rstmgr_causes_bitfield_t;
+
+/**
+ * Reset Manager all possible reset causes enumeration.
+ *
+ * Note: invariants are used to extract information encoded in
+ *       `dif_rstmgr_causes_bitfield_t`, which means that the values must
+ *       correspond to the individual bits (0x1, 0x2, 0x4, ..., 0x80000000).
+ *
+ * TODO: no documentation currently available, would be useful
+ *       for writing a more detailed description for the individual causes.
+ */
+typedef enum dif_rstmgr_cause {
+  kDifRstmgrCausePowerUp = 0x1,
+  kDifRstmgrCauseLowPowerExit = 0x2,
+  kDifRstmgrCauseWatchdog = 0x4,
+  kDifRstmgrCauseSecurityEscalation = 0x8,
+  kDifRstmgrCauseNonDebugModuleRequest = 0x10,
+  kDifRstmgrCauseLast = kDifRstmgrCauseNonDebugModuleRequest,
+} dif_rstmgr_cause_t;
+
+/**
+ * Reset Manager software reset available peripherals.
+ */
+typedef enum dif_rstmgr_peripheral {
+  kDifRstmgrPeripheralSpi = 0,
+  kDifRstmgrPeripheralUsb,
+  kDifRstmgrPeripheralLast = kDifRstmgrPeripheralUsb,
+} dif_rstmgr_peripheral_t;
+
+/**
+ * Reset Manager configuration data.
+ */
+typedef struct dif_rstmgr_config {
+  /**
+   * Reset Manager peripheral software reset lock configuration (`true` to lock,
+   * otherwise nothing, as the enable registers cannot be unlocked).
+   */
+  bool peripheral_software_reset_lock[kDifRstmgrPeripheralLast + 1];
+} dif_rstmgr_config_t;
```

Comment:
I don't think of the reset manager as something you lock at configure-time. Instead provide two methods: one to lock a peripheral, and one to check if a peripheral is locked (i.e. with a bool out-param).

Reply:
Done.